src/test/models/property.spec.ts

```diff
@@ -24,33 +24,48 @@ describe('PropertyTest', () => {
     type: 'bool'
   });
 
-  it('check basic property', () => {
+  const dummyProperty4 = new Property({
+    name: 'cargoDescription',
+    value: 'Statement or description of the cargo.',
+    type: 'str'
+  });
+
+  it('checks basic property', () => {
     expect(dummyProperty1).toBeDefined();
     expect(dummyProperty1.id).toEqual(8);
     expect(dummyProperty1.name).toEqual('cargoDescription');
   });
 
+  it('checks property without id', () => {
+    expect(dummyProperty4.id).toBeUndefined();
+  });
+
   describe('getPostPayload()', () => {
 
     it('checks basic', () => {
-      let dummyPayload = dummyProperty1.getPostPayload();
-      expect(dummyPayload['id']).toEqual(8);
-      expect(dummyPayload['name']).toBe('cargoDescription');
+      let dummyPayload1 = dummyProperty1.getPostPayload();
+      expect(dummyPayload1['id']).toEqual(8);
+      expect(dummyPayload1['name']).toBe('cargoDescription');
+    });
+
+    it('checks basic without id', () => {
+      let dummyPayload4 = dummyProperty4.getPostPayload();
+      expect(dummyPayload4['id']).toBeUndefined();
```

Comment:
I don't believe this is correct. 
The `getPostPayload()` will return `payload['id'] = this.id;` which will result in `payload['id'] = undefined` but when serialized into JSON, will likely return `"payload": null` which will be wrong.
So, the proper check is `expect('id' in dummyPayload4).toBeFalsly`
Very different